redis_commands.c

```diff
@@ -2608,16 +2640,24 @@ static int get_georadius_opts(HashTable *ht, int *withcoord, int *withdist,
     /* Iterate over our argument array, collating which ones we have */
     ZEND_HASH_FOREACH_KEY_VAL(ht, idx, zkey, optval) {
         ZVAL_DEREF(optval);
+
         /* If the key is numeric it's a non value option */
         if (zkey) {
             if (ZSTR_LEN(zkey) == 5 && !strcasecmp(ZSTR_VAL(zkey), "count")) {
                 if (Z_TYPE_P(optval) != IS_LONG || Z_LVAL_P(optval) <= 0) {
-                    php_error_docref(NULL TSRMLS_CC, E_WARNING, "COUNT must be an integer > 0!");
+                    php_error_docref(NULL TSRMLS_CC, E_WARNING,
+                            "COUNT must be an integer > 0!");
+                    if (o->key) zend_string_release(o->key);
                     return FAILURE;
                 }
 
                 /* Set our count */
-                *count = Z_LVAL_P(optval);
+                o->count = Z_LVAL_P(optval);
+            } else if (o->store == STORE_NONE) {
+                o->store = get_georadius_store_type(zkey);
+                if (o->store != STORE_NONE) {
+                    o->key = zval_get_string(optval);
```

Comment:
Probably we need to release previous value before assigning

Reply:
I don't think it can be set twice.  I only enter this block if `o->store == STORE_NONE`.

If we then set the key, setting it again won't happen because `o->store != STORE_NONE`.